File: problems/3159_find_occurrences_of_an_element_in_an_array/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    vector<int> occurrencesOfElement(vector<int>& nums, vector<int>& queries, int x) {
        vector<int> idx;
        for (int i = 0; i < nums.size(); i++) if (nums[i] == x) idx.push_back(i);
        vector<int> res;
        for (int q : queries)
            res.push_back(q <= (int)idx.size() ? idx[q-1] : -1);
        return res;
    }
};

File: problems/3159_find_occurrences_of_an_element_in_an_array/Solution.java
import java.util.*;

class Solution {
    public int[] occurrencesOfElement(int[] nums, int[] queries, int x) {
        List<Integer> idx = new ArrayList<>();
        for (int i = 0; i < nums.length; i++) if (nums[i] == x) idx.add(i);
        int[] res = new int[queries.length];
        for (int i = 0; i < queries.length; i++)
            res[i] = queries[i] <= idx.size() ? idx.get(queries[i]-1) : -1;
        return res;
    }
}
